mermaid-mode.el

```diff
@@ -57,16 +59,31 @@
 
 (defcustom mermaid-mmdc-location "mmdc"
   "Mmdc location."
+  :type 'string
   :group 'mermaid-mode)
 
 (defcustom mermaid-output-format ".png"
   "Mmdc output format."
-  :group 'mermaid-mode)
+  :group 'mermaid-mode
+  :type 'string)
 
 (defconst mermaid-font-lock-keywords
-      '(("graph \\|subgraph \\|end\\|sequenceDiagram\\|loop \\|alt \\|else \\|opt" . font-lock-keyword-face)
-        ("---\\|-?->*\\+?" . font-lock-function-name-face)
-        ("LR\\|TD\\|participant \\|Note" . font-lock-constant-face)))
+      '(("graph \\|subgraph \\|end \\|pie \\|gantt \\|classDiagram \\|stateDiagram \\|title \\|sequenceDiagram\\|loop \\|alt \\|else \\|opt" . font-lock-keyword-face)
+        ("---\\|-?->*\\+?\\|==>\\|===" . font-lock-function-name-face)
+        ("LR\\|TD\\|TB\\|RL\\|DT\\|BT\\|participant \\|Note" . font-lock-constant-face)))
+
+(defvar org-babel-default-header-args:mermaid
+  '((:results . "file") (:exports . "results"))
+  "Default arguments for evaluatiing a mermaid source block.")
+
+(defun org-babel-execute:mermaid (body params)
```

Comment:
how does this work? it is going to be automatically called on org src block with mermaid mode?

Reply:
when you call org-publish-current-file, it will do tranform and replace the block with the image it generated